src/common/mixin/store-change-behaviour.js

```diff
@@ -36,71 +33,85 @@ const changeBehaviourMixin = {
         if (changeInfos && changeInfos.informations && changeInfos.informations.callerId && this._identifier === changeInfos.informations.callerId) {
             return this._displayMessageOnChange(changeInfos);
         }
-
     },
+
     /**
     * Event handler for 'change' events coming from the stores
-    * @param {object} changeInfos - The changing informations.
+    * @param {object} changeInfos The changing informations.
     */
-    _onChange: function onFormStoreChangeHandler(changeInfos) {
-        let onChange = this.props.onChange || this.onChange;
-        if (onChange) {
-            onChange.call(this, changeInfos);
+    _onStoreChange(changeInfos) {
+        const onStoreChange = this.props.onStoreChange || this.onStoreChange;
+        if (onStoreChange) {
+            onStoreChange.call(this, changeInfos);
+        }
+
+        this.setState(this._getStateFromStores(), () => this._afterChange(changeInfos));
+    },
+
+    /**
+    * Read.
+    */
+    _onStoreStatus() {
+        if (this._getEntity) {
+            this.setState({ ...this._getEntity(), ...this._getLoadingStateFromStores() });
+        } else {
+            this.setState(this._getLoadingStateFromStores());
         }
-        this.setState(this._getStateFromStores(changeInfos.property), () => this._afterChange(changeInfos));
     },
+
     /**
     * Event handler for 'error' events coming from the stores.
+    * @param {object} changeInfos The changing informations.
     */
-    _onError: function onFormErrorHandler(changeInfos) {
-        this.setState(this._getLoadingStateFromStores(), () => this._handleErrors(changeInfos)); // update errors after status
+    _onStoreError(changeInfos) {
+        const onStoreError = this.props.onStoreError || this.onStoreError;
+        if (onStoreError) {
+            onStoreError.call(this, changeInfos);
+        }
+
+        this.setState(this._getLoadingStateFromStores(), () => this._handleErrors()); // update errors after status
     },
+
+    /**
+     * Handle errors.
+     */
     _handleErrors() {
         const errorState = this._getErrorStateFromStores();
         if (this.definitionPath) {
             // In case we have a definitionPath, we might want to trigger a setError on the corresponding field
             for (let key in errorState) {
                 // Let's find that corresponding field, considering that the ref might not directly be 'storeNode.fieldName', but in fact 'entityPath.fieldName'
-                const ref = reduce(this.refs, (acc, value, candidateRef) => {
-                    const candidate = candidateRef.replace(`${this.definitionPath}.`, ''); // Remove the 'definitionPath.'
-                    if (candidate === key.match(/([^\.]*)$/)[0]) { // Look for the 'fieldName' part of 'storeNode.fieldName'
-                        acc = value;
+                if (this.refs) {
+                    const refKey = keys(this.refs).filter(candidateRef => {
```

Comment:
If this.refs is an object, I would prefer using native code, like Object.keys.
Same for reduce, if the object is an array, or else it's better to use Array.prototype.reduce.apply.